doc/guides/porting/board_porting.rst

```diff
@@ -45,135 +31,333 @@ available in the source tree:
+named ``plank``:
 
-In order to provide consistency across the various boards and ease the work of
-users providing applications that are not board specific, the following
-guidelines should be followed when porting a board. Unless explicitly stated,
-peripherals should be disabled by default.
+.. code-block:: none
 
-- Configure and enable a working clock configuration, along with a tick source.
+   boards/<ARCH>/plank
+   ├── Kconfig.board
+   ├── Kconfig.defconfig
+   └── plank_defconfig
+
+:file:`Kconfig.board`
+  Included by :zephyr_file:`boards/Kconfig` to include your board
+  in the list of options.
+
+  This should at least contain a definition for a ``BOARD_PLANK`` option,
+  which looks something like this:
+
+  .. code-block:: none
+
+     config BOARD_PLANK
+     	bool "Plank board"
+     	depends on SOC_SERIES_YOUR_SOC_SERIES_HERE
+     	select SOC_PART_NUMBER_ABCDEFGH
+
+:file:`Kconfig.defconfig`
+  Board-specific default values for Kconfig options.
+
+  The entire file should be inside an ``if BOARD_PLANK`` / ``endif`` pair of
+  lines, like this:
+
+  .. code-block:: none
+
+     if BOARD_PLANK
+
+     # Always set CONFIG_BOARD here. This isn't meant to be customized,
+     # but is set as a "default" due to Kconfig language restrictions.
+     config BOARD
+     	default "plank"
+
+     # Other options you want enabled by default go next. Examples:
+
+     config FOO
+     	default y
+
+     if NETWORKING
+     config SOC_ETHERNET_DRIVER
+     	default y
+     endif # NETWORKING
+
+     endif # BOARD_PLANK
+
+:file:`plank_defconfig`
+  A Kconfig fragment that is merged as-is into the final build directory
+  :file:`.config` whenever an application is compiled for your board.
+
+  You should at least select your board's SOC and do any mandatory settings for
+  your system clock, console, etc. The results are architecture-specific, but
+  typically look something like this:
+
+  .. code-block:: none
+
+     CONFIG_SOC_${VENDOR_XYZ3000}=y                      /* select your SoC */
+     CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=120000000   /* set up your clock, etc */
+     CONFIG_SERIAL=y
+
+Build, test, and fix
+********************
+
+Now it's time to build and test the application(s) you want to run on your
+board until you're satisfied.
+
+For example:
+
+.. code-block:: console
+
+   west build -b plank samples/hello_world
+   west flash
+
+For ``west flash`` to work, see :ref:`flash-and-debug-support` below. You can
+also just flash :file:`build/zephyr/zephyr.elf`, :file:`zephyr.hex`, or
+:file:`zephyr.bin` with any other tools you prefer.
+
+General recommendations
+***********************
+
+For consistency and to make it easier for users to build generic applications
+that are not board specific for your board, please follow these guidelines
+while porting.
+
+- Unless explicitly recommended otherwise by this section, leave peripherals
+  and their drivers disabled by default.
```

Comment:
I'm not sure how, but more elaboration on this point might be useful? For a user looking at existing ARM implementations, most of them incorporate things like PWM, I2C, SPI and USB by default. If it is better to leave these disabled for the board definition, maybe it'd be useful to note the proper stage that they should be enabled in the development process?

Reply:
> I'm not sure how, but more elaboration on this point might be useful?
>  For a user looking at existing ARM implementations, most of them incorporate things like PWM, I2C, SPI and USB by default.

Hmm, that seems to be covered below, no?

```
- Provide pin and driver configuration that matches the board's valuable
  components such as sensors, buttons or LEDs, and communication interfaces
  such as USB, Ethernet connector, or Bluetooth/Wi-Fi chip.

- If your board uses a well-known connector standard (like Arduino, Mikrobus,
  Grove, or 96Boards connectors), add connector nodes to your DTS and configure
  pin muxes accordingly.

- Configure components that enable the use of these pins, such as
  configuring an SPI instance to use the usual Arduino SPI pins.
```

> If it is better to leave these disabled for the board definition, maybe it'd be useful to note the proper stage that they should be enabled in the development process?

That is very, very use case specific, I think.

Reply:
I think this could still be elaborated. In the Feather F405 instance, what's the cutoff for supporting a peripheral on a pin? Should PWM only be implemented when the pin has an LED, or even then? Should SPI and I2C be implemented on pins that reference them on the silkscreen? How about pins labeled as Analog pins (A0, A1, etc)? Or should peripherals ONLY be enabled for pins that aren't actually exposed to the user, such as for an on-chip accelerometer? 

Basically, what "counts" for stuff that should be configured at the board level?

Reply:
OK, I see.

I agree this could be improved, but answering this properly would start to get into the zoo of popular development board form factors and their conventions, which I'm not able to get into, so I will leave it to future work.

Might this be something Adafruit would like to help contributing?